File: server-mh/src/Match/Handler/MatchLabel.ts
import { GameMode, TeamMode } from "./Enums";
export class MatchLabel {
    matchStarted = false;
    presentPlayerCount = 0;
    maxPlayers = 4;
    gameMode: GameMode = GameMode.Classic;
    teamMode: TeamMode = TeamMode.None;

    constructor(gameMode: GameMode = GameMode.Classic, teamMode: TeamMode = TeamMode.None) {
        this.gameMode = gameMode;
        this.teamMode = teamMode;
    }
    toJson(): string {
        return JSON.stringify(this);
    }

    static fromJson(json: string): MatchLabel {
        return Object.assign(new MatchLabel(), JSON.parse(json));
    }
    update(dispatcher: nkruntime.MatchDispatcher) {
        dispatcher.matchLabelUpdate(
            JSON.stringify(this)
        );
    }
    static compare(a: string, b: string): number {

        const la = MatchLabel.fromJson(a);
        const lb = MatchLabel.fromJson(b);

        if (la.presentPlayerCount !== lb.presentPlayerCount)
            return lb.presentPlayerCount - la.presentPlayerCount;

        return 0;
    }
}

File: Client-unity/Ludo/Assets/Scripts/OfflineMatchHandler/MatchLabel.cs
namespace Ludo.Offline
{
    public class MatchLabel
    {
        public bool MatchStarted;
        public int PresentPlayerCount;
        public int MaxPlayers = 4;

        public GameMode GameMode = GameMode.Classic;
        public TeamMode TeamMode = TeamMode.None;

        public MatchLabel(
            GameMode gameMode = GameMode.Classic,
            TeamMode teamMode = TeamMode.None)
        {
            GameMode = gameMode;
            TeamMode = teamMode;
        }
    }
}